internal/frrconfig/frrconfig_test.go

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package frrconfig
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+var tests = map[string]struct {
+	failValidate bool
+	failReload   bool
+}{
+	"/tmp/shouldPass": {
+		failValidate: false,
+		failReload:   false,
+	},
+	"/tmp/failValidate": {
+		failValidate: true,
+		failReload:   false,
+	},
+	"/tmp/failReload": {
+		failValidate: false,
+		failReload:   true,
+	},
+}
+
+func TestReload(t *testing.T) {
+	execCommand = fakeExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	for tc, params := range tests {
+		t.Run(fmt.Sprintf("reload %s", tc), func(t *testing.T) {
+			err := Update(tc)
+			if (params.failReload || params.failValidate) && err == nil {
+				t.Fatalf("expecting failure, got no error")
+			}
+			if params.failReload && !strings.Contains(err.Error(), "reload") {
+				t.Fatalf("expecting reload error, got %v", err)
+			}
+			if params.failValidate && !strings.Contains(err.Error(), "test") {
+				t.Fatalf("expecting test error, got %v", err)
+			}
+			if !params.failReload && !params.failValidate && err != nil {
+				t.Fatalf("expecting no error, got %v", err)
+			}
+		})
+	}
+}
+
+// helper function that redirects the execution to a mock process implemented by
+// TestHelperProcess
+func fakeExecCommand(name string, args ...string) *exec.Cmd {
+	cs := []string{"-test.run=TestFakeReloadHelper", "--"}
+	cs = append(cs, args...)
+	env := []string{
+		"WANT_FAKE_PYTHON=true",
+	}
+
+	cmd := exec.Command(os.Args[0], cs...)
+	cmd.Env = append(env, os.Environ()...)
+	return cmd
+}
+
+// This is not a real test. It's used in case fakeExecCommand is used in place of exec.Command.
+// In that case the command execution is redirected to this function.
+func TestFakeReloadHelper(t *testing.T) {
```

Comment:
seems like a bit of over-machinery to make CombinedOutput happy (iiuc), can't execCommand be more like `exec.Exec` instead of `exec.Command` ?

Reply:
Not sure I understood your comment. Anyhow, this kind of test is heavily inspired (ie, blatantly copied) from this talk https://youtu.be/8hQG7QlcLBk?t=2052

Basically the idea is, to test the interaction with a sub process, we run the very same executable (in this case the test one) in such a way that instead of running a test (TestFakeReloadHelper), it behaves as a test double.
With this machinery in place, we allow the test (the double) to intercept what the real program argument would be.

Reply:
sorry let me expand what I meant 😅 
currently the `execCommand` var (that we test and replace) is a `func(name string, arg ...string) *exec.Cmd`, which is what caused the introduction of this `TestFakeReloadHelper` approach (and the whole machinery).
since we don't really use what `exec.Cmd` has to offer (other than CombinedOutput) the `execCommand` var could just be a `func(name string, arg ...string) (string,error)` which is a lot more straight-forward to test imo (and what we usually did so far).

Reply:
so iiuc, you are a suggesting not to sub process and interact with the external process through the machinery, but just to mock the command execution itself.